test/conformance/program/urProgramSetSpecializationConstants.cpp

```diff
@@ -31,6 +43,88 @@ TEST_P(urProgramSetSpecializationConstantsTest, Success) {
     ValidateBuffer<uint32_t>(buffer, sizeof(spec_value), spec_value);
 }
 
+TEST_P(urProgramSetSpecializationConstantsTest, UseDefaultValue) {
+    ur_platform_backend_t backend;
+    ASSERT_SUCCESS(urPlatformGetInfo(platform, UR_PLATFORM_INFO_BACKEND,
+                                     sizeof(ur_platform_backend_t), &backend,
+                                     nullptr));
+    if (backend == UR_PLATFORM_BACKEND_CUDA) {
+        GTEST_FAIL()
+            << "This test is known to cause crashes on Nvidia; not running.";
+    }
+
+    ASSERT_SUCCESS(urProgramBuild(context, program, nullptr));
+    auto entry_points =
+        uur::KernelsEnvironment::instance->GetEntryPointNames(program_name);
+    kernel_name = entry_points[0];
+    ASSERT_SUCCESS(urKernelCreate(program, kernel_name.data(), &kernel));
+
+    ur_mem_handle_t buffer;
+    AddBuffer1DArg(sizeof(spec_value), &buffer);
+    Launch1DRange(1);
+    ValidateBuffer<uint32_t>(buffer, sizeof(spec_value), default_spec_value);
+}
+
+TEST_P(urProgramSetMultipleSpecializationConstantsTest, MultipleCalls) {
+    uint32_t a = 100;
+    uint64_t b = 200;
+    bool c = false;
+    uint64_t output = 0;
+
+    ur_specialization_constant_info_t info_a = {0, sizeof(uint32_t), &a};
+    ASSERT_SUCCESS(urProgramSetSpecializationConstants(program, 1, &info_a));
+
+    ur_specialization_constant_info_t info_c = {2, sizeof(bool), &c};
+    ASSERT_SUCCESS(urProgramSetSpecializationConstants(program, 1, &info_c));
+
+    ur_specialization_constant_info_t info_b = {1, sizeof(uint64_t), &b};
+    ASSERT_SUCCESS(urProgramSetSpecializationConstants(program, 1, &info_b));
```

Comment:
any reason for spec id 2 coming before spec id 1?

Reply:
Just a small sanity check to ensure "out of order" sets work correctly. This would catch a naive implementation that assumed that `urProgramSetSpecializationConstant`s were always called in order.